third_party/move/move-binary-format/src/deserializer.rs

```diff
@@ -13,7 +13,7 @@ use std::{collections::HashSet, convert::TryInto, io::Read};
 impl CompiledScript {
     /// Deserializes a &[u8] slice into a `CompiledScript` instance.
     pub fn deserialize(binary: &[u8]) -> BinaryLoaderResult<Self> {
-        let config = DeserializerConfig::new(VERSION_DEFAULT, IDENTIFIER_SIZE_MAX);
+        let config = DeserializerConfig::new(VERSION_MAX, IDENTIFIER_SIZE_MAX);
```

Comment:
If you want the default to become VERSION_MAX, you should update the constant VERSION_DEFAULT. Its not unexpected that for some time those constants have the same value. Later then, VERSION_MAX will become 8.

Reply:
@wrwg VERSION_DEFAULT was introduced to distinguish V6 and V7 for resource access control. I am wondering after we enable V7 feature flag, do we want to set both of them to 7?

Reply:
The general picture is (which I setup a while ago):

- VERSION_DEFAULT is the highest version we run on any chain.
- VERSION_MAX can be one higher when we are currently developing a new version, as now.
- When we ship the new version (as now) we establish VERSION_DEFAULT == VERSION_MAX. After a while, when we work on the _next_ file format version v8, we may increment VERSION_MAX=8.